src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -46,24 +46,35 @@ private void RegisterCommandCallback()
 
         private void OnEventSourceCommand(object? sender, EventCommandEventArgs e)
         {
-            if (e.Command == EventCommand.Enable || e.Command == EventCommand.Update)
-            {
-                Debug.Assert(e.Arguments != null);
+            // Should only be enable or disable
+            Debug.Assert(e.Command == EventCommand.Enable || e.Command == EventCommand.Disable);
 
-                if (e.Arguments.TryGetValue("EventCounterIntervalSec", out string? valueStr) && float.TryParse(valueStr, out float value))
+            lock (s_counterGroupLock)      // Lock the CounterGroup
+            {
+                float intervalValue = 1.0f;
```

Comment:
I might be misunderstanding... because this has a non-zero value by default, isn't this going to automatically initialize counters at 1000ms intervals for the very first enabled event source if it doesn't have `EventCounterIntervalSec` in its spec? Won't that prevent any real request for counters to ever be observed?

Reply:
It's definitely wrong, good catch. As it is we will always enable counters for any Enable command. We would process future Enable commands though so it probably wouldn't impact anyone other than some additional overhead in the process